.github/workflows/pytest.yml

```diff
@@ -0,0 +1,38 @@
+name: Pytest and Black formatting
+
+on:
+    - pull_request
+    - push
```

Comment:
Should we just narrow the scope of push to only main branch? with current configuration I think it will trigger the workflow  on every branch on every push?

Reply:
We would still want tests for branches regardless?